api/trace/api.go

```diff
@@ -137,13 +138,25 @@ type Link struct {
 	Attributes []core.KeyValue
 }
 
+// SpanKind represents the role of a Span inside a Trace. Often, this defines how a Span
+// will be processed and visualized by various backends.
+type SpanKind int
```

Comment:
I would prefer to make this type `string`. For the instances below, use the conventional name "server", "client", ...

I suggest this because in OpenTracing, instead of an explicit  field in the span structure, a `span.kind` attribute was specified. In an OpenTracing bridge, we can just use `SpanKind(value)` and won't have to construct an explicit switch statement.

I think it's an added benefit that users can specify unconventional span kinds as well, although some will disagree. I see this value as an advisory statement, helpful to vendors, but we shouldn't force all span kinds to be one of these four values.

Reply:
I took the bullet here and changed it to string and implemented the bridge for SpanKind. I think this is fine for now, but we should go to specs to clarify the behavior of the `SpanKind`, java implementation is an `Enum` and only accept the values on the spec and any other value goes to `internal`. 

WDYT?

Reply:
Thank you. 😄
I think we should do it, and I'll agree to back this up by filing an issue in the spec repo, sometime this week.